ocaml/xapi/xapi_xenops.ml

```diff
@@ -671,6 +671,34 @@ module MD = struct
         )
         [] vm.API.vM_VGPUs
 
+  let of_vusb ~__context ~vm ~pusb =
+    let open Vusb in
+    try
+      let path = pusb.API.pUSB_path in
+      let pathList= Xstringext.String.split '-' path in
+      let hostbus = List.nth pathList 0 in
+      let hostport = List.nth pathList 1 in
+      (* Here version can be 1.10/2.00/3.00. *)
+      let version = pusb.API.pUSB_version in
+      {
+        id = (vm.API.vM_uuid, "vusb"^path);
+        hostbus = hostbus;
+        hostport = hostport;
+        version = version;
+        path = path;
+      }
+    with
+      | _ -> failwith "PUSB settings invalid"
```

Comment:
Don't use `failwith` but `error` instead. Include `path` in the message to make it more useful.

Reply:
updated.